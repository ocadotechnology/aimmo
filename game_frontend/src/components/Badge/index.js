import React, { Component } from 'react'
import styled from 'styled-components'

import Box from '@material-ui/core/Box'
import Modal from '@material-ui/core/Modal'
import Typography from '@material-ui/core/Typography'

import { badgeInfo } from './badges'

const BadgeModalBox = styled(Box)`
  position: absolute;
  top: 15%;
  left: 65%;
  width: 400px;
  background-color: #fff;
  border-radius: 10px;
  padding: 25px;
  opacity: 0.8;
`

const BadgeModalImg = styled(Box)`
  height: 55px;
  position: absolute;
  top: 35px;
  right: 25px;
`

export default class BadgeModal extends Component {
  render() {
    if (!this.props.modalOpen) {
      return null
    }

    const taskId = this.props.taskId
    const info = badgeInfo[taskId]

    return (
      <Modal
        open={true}
        hideBackdrop={true}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <BadgeModalBox>
          <Box style={{ width: 330 }}>
            <Typography variant="h6">{info.title}</Typography>
            <Typography variant="subtitle1">{info.message}</Typography>
          </Box>
          <BadgeModalImg component="img" alt={info.name} src={info.img}></BadgeModalImg>
        </BadgeModalBox>
      </Modal>
    )
  }
}

export function getBadges(tasks) {
  return tasks.map((task) => (
    <Box
      component="img"
      style={{ height: 45, marginRight: 15 }}
      alt={badgeInfo[task].name}
      title={badgeInfo[task].name}
      src={badgeInfo[task].img}
      key={task}
    ></Box>
  ))
}
