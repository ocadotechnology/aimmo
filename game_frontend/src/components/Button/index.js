import styled from 'styled-components'

const Button = styled.button`
  padding: 10px
  background-color: #3886AA
  border-radius: 15px
  margin: 10px
  color: white

  &:hover {
    background-color: #2E6E8C
  }
`
export default Button
