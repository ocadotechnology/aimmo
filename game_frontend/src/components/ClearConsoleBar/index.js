import { Button, Toolbar } from '@material-ui/core';
import ClearIcon from 'components/icons/Clear';
import React from 'react';
import styled from 'styled-components'
import PropTypes from 'prop-types'

export const StyledClearConsoleBar = styled(Toolbar)`
  background: #2f3129;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`

export const StyledConsoleTitle = styled.div`
  display: inline-flex;
`

export const ClearButton = styled(Button)`
  right: 0vw;
`

const ClearConsoleBar = (props) => {
  return (
    <StyledClearConsoleBar variant="dense">
      <StyledConsoleTitle>
        <ClearButton
          variant="outlined"
          onClick={props.clearConsoleClicked}
          startIcon={<ClearIcon />}
        >
          Clear console
        </ClearButton>
      </StyledConsoleTitle>
    </StyledClearConsoleBar>
  )
}

ClearConsoleBar.propTypes = {
  clearConsoleClicked: PropTypes.func,
}

export default ClearConsoleBar;
