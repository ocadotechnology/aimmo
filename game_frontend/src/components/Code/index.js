import styled from 'styled-components'

const Code = styled.p`
  font-family: ${props =>
    props.theme.additionalVariables.typography.code.fontFamily};
  color: ${props => props.theme.palette.primary.main};
`
export default Code
