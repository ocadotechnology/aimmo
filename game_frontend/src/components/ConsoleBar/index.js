import styled from 'styled-components'
import Toolbar from '@material-ui/core/Toolbar/Toolbar'
import { Typography } from '@material-ui/core'
import ConsoleIcon from 'components/icons/Console'
import React, { Component } from 'react'

export const StyledConsoleBar = styled(Toolbar)`
  background-color: ${props => props.theme.palette.grey['A700']};
`

export const StyledConsoleIcon = styled(ConsoleIcon)`
  padding-right: ${props => props.theme.spacing()}px;
`

export default class ConsoleBar extends Component {
  render () {
    return (
      <StyledConsoleBar variant='dense' >
        <StyledConsoleIcon color='disabled' />
        <Typography
          variant='body1'
          color='textSecondary'>
          Console Log
        </Typography>
      </StyledConsoleBar>
    )
  }
}
