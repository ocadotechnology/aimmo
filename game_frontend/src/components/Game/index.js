import React, { Component, Fragment } from 'react'
import GameMenu from 'components/GameMenu'
import GameView from 'components/GameView'

export default class Game extends Component {
  render () {
    return (
      <Fragment>
        <GameMenu />
        <GameView />
      </Fragment>
    )
  }
}
