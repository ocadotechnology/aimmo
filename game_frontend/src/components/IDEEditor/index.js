import styled from 'styled-components'
import React, { PureComponent } from 'react'
import AceEditor from 'react-ace'
import 'brace/theme/idle_fingers'
import 'brace/mode/python'
import 'brace/snippets/python'
import 'brace/ext/language_tools'
import PropTypes from 'prop-types'
import { withTheme } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import PlayIcon from 'components/icons/Play'

export const IDEEditorLayout = styled.div`
  background-color: #2F4F4F;
  position: relative;
  grid-area: ide-editor;
`

export const RunCodeButton = styled(Button)`
  && {
    position: absolute;
    right: ${props => props.theme.spacing.unit * 3}px;
    bottom: ${props => props.theme.spacing.unit * 3}px;
    z-index: 5;
  }
`

export const MarginedPlayIcon = styled(PlayIcon)`
  margin-right: ${props => props.theme.spacing.unit}px;
`

export class IDEEditor extends PureComponent {
  render () {
    return (
      <IDEEditorLayout>
        <AceEditor
          mode='python'
          theme='idle_fingers'
          name='ace_editor'
          onLoad={this.props.getCode}
          onChange={this.props.editorChanged}
          fontSize={this.props.theme.additionalVariables.typography.code.fontSize}
          showPrintMargin
          showGutter
          highlightActiveLine
          value={this.props.code}
          width='100%'
          height='100%'
          setOptions={{
            enableBasicAutocompletion: true,
            enableLiveAutocompletion: true,
            enableSnippets: true,
            showLineNumbers: true,
            tabSize: 2,
            fontFamily: this.props.theme.additionalVariables.typography.code.fontFamily
          }} />
        <RunCodeButton
            aria-label='Run Code'
            variant='extendedFab'
            id='post-code-button'
            onClick={this.props.postCode}>
            <MarginedPlayIcon />Run Code
        </RunCodeButton>
      </IDEEditorLayout>
    )
  }
}

IDEEditor.propTypes = {
  code: PropTypes.string,
  getCode: PropTypes.func,
  editorChanged: PropTypes.func,
  theme: PropTypes.object,
  postCode: PropTypes.func
}

export default withTheme()(IDEEditor)
