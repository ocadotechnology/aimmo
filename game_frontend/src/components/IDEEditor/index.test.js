/* eslint-env jest */
import React from 'react'
import { IDEEditor, IDEEditorLayout, MarginedPlayIcon, RunCodeButton } from 'components/IDEEditor'
import createShallowWithTheme from 'testHelpers/createShallow'

describe('<IDEEditor />', () => {
  it('matches snapshot', () => {
    const props = {
      code: 'class Avatar',
      getCode: jest.fn(),
      postCode: jest.fn()
    }

    const component = createShallowWithTheme(<IDEEditor {...props} />)

    expect(component).toMatchSnapshot()
  })

  it('calls the postCode function in props when Post code button is pressed', () => {
    const postCode = jest.fn()
    const props = {
      postCode
    }

    const component = createShallowWithTheme(<IDEEditor {...props} />)

    component.find('#post-code-button').simulate('click')
    expect(postCode).toBeCalled()
  })
})

describe('<IDEEditorLayout />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<IDEEditorLayout />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<RunCodeButton />', () => {
  it('matches snapshot', () => {
    const component = createShallowWithTheme(<RunCodeButton />)

    expect(component).toMatchSnapshot()
  })
})

describe('<MarginedPlayIcon />', () => {
  it('matches snapshot', () => {
    const component = createShallowWithTheme(<MarginedPlayIcon />)

    expect(component).toMatchSnapshot()
  })
})
