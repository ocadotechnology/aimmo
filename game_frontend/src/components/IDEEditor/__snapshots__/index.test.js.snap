// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IDEEditor /> matches snapshot 1`] = `
<styled.div>
  <ReactAce
    cursorStart={1}
    editorProps={Object {}}
    enableBasicAutocompletion={false}
    enableLiveAutocompletion={false}
    focus={false}
    fontSize="1rem"
    height="100%"
    highlightActiveLine={true}
    maxLines={null}
    minLines={null}
    mode="python"
    name="ace_editor"
    onChange={null}
    onLoad={[MockFunction]}
    onPaste={null}
    onScroll={null}
    readOnly={false}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={
      Object {
        "enableBasicAutocompletion": true,
        "enableLiveAutocompletion": true,
        "enableSnippets": true,
        "fontFamily": "'Source Code Pro', monospace",
        "showLineNumbers": true,
        "tabSize": 2,
      }
    }
    showGutter={true}
    showPrintMargin={true}
    style={Object {}}
    tabSize={4}
    theme="idle_fingers"
    value="class Avatar"
    width="100%"
    wrapEnabled={false}
  />
  <Styled(WithStyles(Button))
    aria-label="Run Code"
    id="post-code-button"
    onClick={[MockFunction]}
    variant="extendedFab"
  >
    <Styled(PlayIcon) />
    Run Code
  </Styled(WithStyles(Button))>
</styled.div>
`;

exports[`<IDEEditorLayout /> renders correctly 1`] = `
.c0 {
  background-color: #2F4F4F;
  position: relative;
  grid-area: ide-editor;
}

<div
  className="c0"
/>
`;

exports[`<MarginedPlayIcon /> matches snapshot 1`] = `
.c0 {
  margin-right: 8px;
}

<PlayIcon
  className="c0"
  theme={
    Object {
      "additionalVariables": Object {
        "typography": Object {
          "code": Object {
            "fontFamily": "'Source Code Pro', monospace",
            "fontSize": "1rem",
          },
        },
      },
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {},
      "palette": Object {
        "action": Object {
          "active": "#fff",
          "disabled": "rgba(255, 255, 255, 0.3)",
          "disabledBackground": "rgba(255, 255, 255, 0.12)",
          "hover": "rgba(255, 255, 255, 0.1)",
          "hoverOpacity": 0.1,
          "selected": "rgba(255, 255, 255, 0.2)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#303030",
          "paper": "#424242",
        },
        "common": Object {
          "black": "#000",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(255, 255, 255, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "primary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(178, 178, 178)",
          "light": "rgb(255, 255, 255)",
          "main": "#FFFFFF",
        },
        "secondary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(144, 143, 137)",
          "light": "rgb(216, 215, 208)",
          "main": "#CFCDC5",
        },
        "text": Object {
          "disabled": "rgba(255, 255, 255, 0.5)",
          "hint": "rgba(255, 255, 255, 0.5)",
          "icon": "rgba(255, 255, 255, 0.5)",
          "primary": "#fff",
          "secondary": "rgba(255, 255, 255, 0.7)",
        },
        "tonalOffset": 0.2,
        "type": "dark",
      },
      "props": Object {},
      "shadows": Array [
        "none",
        "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
        "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
        "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
        "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
        "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
        "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
        "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
        "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
        "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
        "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
        "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
        "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
        "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
        "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
      ],
      "shape": Object {
        "borderRadius": "24px",
      },
      "spacing": Object {
        "unit": 8,
      },
      "text": Object {
        "secondary": "#FFFFFF",
      },
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.46429em",
        },
        "body2": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.71429em",
        },
        "button": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "textTransform": "capitalize",
        },
        "caption": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.75rem",
          "fontStyle": "italic",
          "fontWeight": 500,
          "lineHeight": "1.375em",
        },
        "display1": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "2.125rem",
          "fontWeight": 700,
          "lineHeight": "1.20588em",
        },
        "display2": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "2.8125rem",
          "fontWeight": 700,
          "lineHeight": "1.13333em",
          "marginLeft": "-.02em",
        },
        "display3": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "3.5rem",
          "fontWeight": 900,
          "letterSpacing": "-.02em",
          "lineHeight": "1.30357em",
          "marginLeft": "-.02em",
        },
        "display4": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "7rem",
          "fontWeight": 900,
          "letterSpacing": "-.04em",
          "lineHeight": "1.14286em",
          "marginLeft": "-.04em",
        },
        "fontFamily": "museo-sans,Source Code Pro",
        "fontSize": 14,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "headline": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": "1.35417em",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subheading": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": "1.5em",
        },
        "title": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1.3125rem",
          "fontWeight": 500,
          "lineHeight": "1.16667em",
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
/>
`;

exports[`<RunCodeButton /> matches snapshot 1`] = `
.c0.c0 {
  position: absolute;
  right: 24px;
  bottom: 24px;
  z-index: 5;
}

<WithStyles(Button)
  className="c0"
  theme={
    Object {
      "additionalVariables": Object {
        "typography": Object {
          "code": Object {
            "fontFamily": "'Source Code Pro', monospace",
            "fontSize": "1rem",
          },
        },
      },
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {},
      "palette": Object {
        "action": Object {
          "active": "#fff",
          "disabled": "rgba(255, 255, 255, 0.3)",
          "disabledBackground": "rgba(255, 255, 255, 0.12)",
          "hover": "rgba(255, 255, 255, 0.1)",
          "hoverOpacity": 0.1,
          "selected": "rgba(255, 255, 255, 0.2)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#303030",
          "paper": "#424242",
        },
        "common": Object {
          "black": "#000",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(255, 255, 255, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "primary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(178, 178, 178)",
          "light": "rgb(255, 255, 255)",
          "main": "#FFFFFF",
        },
        "secondary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(144, 143, 137)",
          "light": "rgb(216, 215, 208)",
          "main": "#CFCDC5",
        },
        "text": Object {
          "disabled": "rgba(255, 255, 255, 0.5)",
          "hint": "rgba(255, 255, 255, 0.5)",
          "icon": "rgba(255, 255, 255, 0.5)",
          "primary": "#fff",
          "secondary": "rgba(255, 255, 255, 0.7)",
        },
        "tonalOffset": 0.2,
        "type": "dark",
      },
      "props": Object {},
      "shadows": Array [
        "none",
        "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
        "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
        "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
        "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
        "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
        "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
        "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
        "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
        "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
        "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
        "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
        "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
        "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
        "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
      ],
      "shape": Object {
        "borderRadius": "24px",
      },
      "spacing": Object {
        "unit": 8,
      },
      "text": Object {
        "secondary": "#FFFFFF",
      },
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.46429em",
        },
        "body2": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": "1.71429em",
        },
        "button": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "textTransform": "capitalize",
        },
        "caption": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "0.75rem",
          "fontStyle": "italic",
          "fontWeight": 500,
          "lineHeight": "1.375em",
        },
        "display1": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "2.125rem",
          "fontWeight": 700,
          "lineHeight": "1.20588em",
        },
        "display2": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "2.8125rem",
          "fontWeight": 700,
          "lineHeight": "1.13333em",
          "marginLeft": "-.02em",
        },
        "display3": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "3.5rem",
          "fontWeight": 900,
          "letterSpacing": "-.02em",
          "lineHeight": "1.30357em",
          "marginLeft": "-.02em",
        },
        "display4": Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "7rem",
          "fontWeight": 900,
          "letterSpacing": "-.04em",
          "lineHeight": "1.14286em",
          "marginLeft": "-.04em",
        },
        "fontFamily": "museo-sans,Source Code Pro",
        "fontSize": 14,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "headline": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": "1.35417em",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subheading": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": "1.5em",
        },
        "title": Object {
          "color": "#fff",
          "fontFamily": "museo-sans,Source Code Pro",
          "fontSize": "1.3125rem",
          "fontWeight": 500,
          "lineHeight": "1.16667em",
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
/>
`;
